Improvements to Settings Page UI, Backend Integration, Alertmanager Addition

lwu15 opened the pull request:
- Refactored the Settings page UI and integrated it with the backend using routers. Applied styling to the Settings page.
- Added Alertmanager to Docker Compose and updated Prometheus configuration.
- Updated Docker Compose to mount Alertmanager and successfully send alerts.
- Removed the URL from alertmanager.yml and stored it separately.

Testing and Configuration: Code has been locally tested with docker compose up for altering. When testing, be sure to update the Slack API URL in the alertmanager.yml file with the actual URL before running the docker compose up command. (Note that the URL is invalidated once included in a commit). Then go to the localhost for prometheus and alertmanager respectively. 

Also, successfully tested that alert preferences are stored in the database.

lwu15 committed:
add test local storage

lwu15 committed:
Revert "add test local storage"

This reverts commit c52f1977f00c9c5eba27ab2bd20f9630103ed190.

lwu15 committed:
Refactor settings page UI and integrate with back end using routers

lwu15 committed:
Add Alertmanager to Docker Compose, Update Prometheus Configuration

lwu15 committed:
Update Docker Compose to mount Alertmanager, now successfully sends Slack notifications

lwu15 committed:
Implement env variable substitution in Electron app for dynamic config loading

lwu15 committed:
styling for settings

lwu15 committed:
add comments for Settings, comment out some code related to connecting alertmanager to back end endpoint

lwu15 committed:
remove url from alertmanager.yml, store seperately

lwu15 committed:
minor edits in alertmanager.yml

lwu15 committed:
Merge branch 'dev' into lwu/alerts

echochs1 commented on src/pages/Settings.jsx:
```diff
@@ -1,128 +1,256 @@
+      return;
+    }
+
+    if (!password.trim()) {
```
Doesn't check entered password with hashed password in DB

lwu15 replied:
updated

lwu15 committed:
Add password verification for updating settings and update authControllers error propagation

lwu15 committed:
comment out console.log's

lwu15 committed:
Merge branch 'dev' into lwu/alerts

lwu15 commented:
@echochs1  Elliot, I appreciate your code review. Following our discussion, I've made the following changes:

- Removed `JSON.stringify` for password comparison since it was duplicative. Also, I'm now verifying users on the backend/added password verification for updating settings.
- - Planning to streamline the Settings page UI for a better user experience.

Additionally, updated `authControllers` to improve error propagation. Previously, it was not using `next` for the error messages/sending them directly but now correctly sends the error messages for the login page.

I've also commented out unnecessary `console.log` statements.

Ready for your review and feedback.

echochs1 reviewed (approved):
Looks great!
